src/AppInstallerCommonCore/Public/winget/FolderFileWatcher.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "pch.h"
+
+namespace AppInstaller::Utility
+{
+    // Watch for new/renamed files recursively in a given directory.
+    struct FolderFileWatcher

```

Comment:
Isn't a `FolderFileWatcher` just a `FolderFileExtensionWatcher` with a `null_opt` optional extension to look for?